packetery/src/Packetery/Order/GridExtender.php

```diff
@@ -0,0 +1,234 @@
+	 * @param Engine     $latteEngine Latte Engine.
+	 * @param Request    $httpRequest Http Request.
+	 */
+	public function __construct(
+		Helper $helper,
+		Repository $carrierRepository,
+		Engine $latteEngine,
+		Request $httpRequest
+	) {
+		$this->helper            = $helper;
+		$this->carrierRepository = $carrierRepository;
+		$this->latteEngine       = $latteEngine;
+		$this->httpRequest       = $httpRequest;
+	}
+
+	/**
+	 * Adds custom filtering links to order grid.
+	 *
+	 * @param array $var Array of html links.
+	 *
+	 * @return array
+	 */
+	public function addFilterLinks( array $var ): array {
+		$orders      = wc_get_orders( [ 'packetery_to_submit' => '1' ] );
+		$latteParams = [
+			'link'       => add_query_arg(
+				[
+					'packetery_to_submit' => '1',
+					'packetery_to_print'  => false,
+				]
+			),
+			'title'      => __( 'packetaOrdersToSubmit', 'packetery' ),
+			'orderCount' => count( $orders ),
+		];
+		$var[]       = $this->latteEngine->renderToString( PACKETERY_PLUGIN_DIR . '/template/order/filter-link.latte', $latteParams );
+
+		$orders      = wc_get_orders( [ 'packetery_to_print' => '1' ] );
+		$latteParams = [
+			'link'       => add_query_arg(
+				[
+					'packetery_to_submit' => false,
+					'packetery_to_print'  => '1',
+				]
+			),
+			'title'      => __( 'packetaOrdersToPrint', 'packetery' ),
+			'orderCount' => count( $orders ),
+		];
+		$var[]       = $this->latteEngine->renderToString( PACKETERY_PLUGIN_DIR . '/template/order/filter-link.latte', $latteParams );
+
+		return $var;
+	}
+
+	/**
+	 * Adds select to order grid.
+	 */
+	public function addSelect(): void {
+		$this->latteEngine->render(
+			PACKETERY_PLUGIN_DIR . '/template/order/type-select.latte',
+			[
+				'packeteryOrderType' => $this->httpRequest->getQuery( 'packetery_order_type' ),
+			]
+		);
+	}
+
+	/**
+	 * Adds query vars to order list request.
+	 *
+	 * @param array $queryVars Query vars.
+	 *
+	 * @return array
+	 */
+	public function addQueryVarsToRequest( array $queryVars ): array {
+		global $typenow;
+
+		if ( in_array( $typenow, wc_get_order_types( 'order-meta-boxes' ), true ) ) {
+			$queryVars = $this->addQueryVars( $queryVars, $this->httpRequest->getQuery() );
+		}
+
+		return $queryVars;
+	}
+
+	/**
+	 * Adds query vars to fetch order list.
+	 *
+	 * @param array $queryVars Query vars.
+	 * @param array $get Get parameters.
+	 *
+	 * @return array
+	 */
+	public function addQueryVars( array $queryVars, array $get ): array {
+		if ( ! empty( $get['packetery_to_submit'] ) ) {
+			$queryVars['meta_query'] = array(
+				array(
+					'key'     => 'packetery_carrier_id',
+					'value'   => '',
+					'compare' => '!=',
+				),
+				array(
+					'key'     => 'packetery_is_exported',
+					'value'   => '',
+					'compare' => '=',
+				),
+			);
+		}
+
+		if ( ! empty( $get['packetery_to_print'] ) ) {
+			$queryVars['meta_query'] = array(
+				array(
+					'key'     => 'packetery_packet_id',
```

Comment:
Nesouvisí to s meta daty objednávky? Pokud ano tak je to tu už dvakrát. Tuším, že jsem to viděl v Checkout třídě. Pokud je to tu už 2x+ , tak do konstanty.